sdk/state/ingest.go

```diff
@@ -273,9 +273,11 @@ func (c *Channel) ingestFormationTx(tx *txnbuild.Transaction, resultXDR string,
 	}
 
 	// Validate the initiator escrow account sequence number is correct.
-	if int64(initiatorEscrowAccountEntry.SeqNum) != c.startingSequence {
+	fmt.Println("meta", initiatorEscrowAccountEntry.SeqNum)
+	fmt.Println("formation", formationTx.SequenceNumber())
+	if int64(initiatorEscrowAccountEntry.SeqNum) != formationTx.SequenceNumber() {
```

Comment:
💭  it _should_ always be the same value but thoughts on pulling the sequence number from the channel.openAgreement? To get it from the source

Reply:
We could pull it from either place. Worth noting that `formationTx` is not the `tx` being passed in to be ingested, it's the tx that `OpenTx` creates so it is based on the internal state of the channel and not an external source, so we're still comparing the external data against purely internal data.

If we pulled it from the open agreement we'd be pulling the `openAgreement.Details.StartingSequence`. As it happens the `formationTx` sequence is set based off that value, so I think comparing them against one or the other is much the same. Something I like about pulling it from the `formationTx` is that on Stellar the rule is that if a tx is executed the account becomes the account's sequence number, so this check is really guaranteeing that has occurred.

Reply:
ya  I was thinking since the formationTx gets the sequenceNum from the open agreement, so the original source. But formationTx sgtm 👍 

but jw what do you mean by this?
> if a tx is executed the account becomes the account's sequence number

Reply:
>what do you mean by this?

That's how the Stellar protocol works. When a tx is executed The source accounts sequence number becomes the sequence number of the tx. It just happens account sequence numbers also have to be sequential, but that requirement will be lifted in cap21.